pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -368,17 +368,15 @@ protected boolean createColumnV1Indices(String column)
           for (String argument : arguments) {
             ColumnMetadata columnMetadata = _segmentMetadata.getColumnMetadataFor(argument);
             if (columnMetadata == null) {
-              LOGGER.warn("Skip creating derived column: {} because argument: {} does not exist in the segment", column,
-                  argument);
-              if (errorOnFailure) {
-                throw new RuntimeException(String.format(
-                    "Failed to create derived column: %s because argument: %s does not exist in the segment", column,
-                    argument));
-              }
-              return false;
-            }
-            // TODO: Support creation of derived columns from forward index disabled columns
-            if (!_segmentWriter.hasIndexFor(argument, StandardIndexes.forward())) {
+              // replace the transform function with the default value for the derived column
+              String expression = _schema.getFieldSpecFor(column).getDefaultNullValueString();
```

Comment:
You can simply short circuit it by invoking `createDefaultValueColumnV1Indices(column)`

Reply:
Yes, much better. Thanks for pointing it out!
reverted the changes and did as you suggested.